func_decor.py

```diff
@@ -0,0 +1,64 @@
+from time import time
+
+registry = {}
+
+
+def wrapper(func):
+    def timer(*args, **kwargs):
+        start = time()
+        res = func(*args, **kwargs)
+        stop = time()
+        registry[func.__name__] = (
+            "result: {result}; function execution time: {timer}".format
+            (result=res, timer=stop - start))
+        return res
+
+    return timer
+
+
+@wrapper
+def spam(*args):
+    return {arg: arg ** 2 for arg in args}  # spam = wrapper(spam)
+
+
+@wrapper
+def get_ranges(lst):
+    """Function "get_ranges".
+
+    This function takes list of unique sorted ascending integer numbers
+    and returns this list rolled up.
+    For example:
+    get_ranges([0, 1, 2, 3, 4, 7, 8, 10]) // "0-4,7-8,10"
+    """
+    pos = 0
+    rolled_list = str(lst[0])
```

Comment:
am I correct you name string variable list?

Reply:
Yep. Y answered in telegram some days ago that output should be a string. The task was to roll out the list.
I'll correct to "roll".